experimental/oak_baremetal_kernel/src/args.rs

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use atomic_refcell::AtomicRefCell;
+use core::ffi::CStr;
+use lazy_static::lazy_static;
+
+// This is a really crude analogue of CString, which we can't use, as CString requires allocation.
+// Instead of allocation, we rely on a static buffer.
+struct StaticCString {
+    buf: [u8; 512],
+
+    // Index of \0 in the buffer, above, so that we don't have to keep searching for it.
+    len: usize,
+}
+
+impl Default for StaticCString {
+    fn default() -> Self {
+        StaticCString {
+            buf: [0; 512],
+            len: 0,
+        }
+    }
+}
+
+impl StaticCString {
+    fn as_c_str(&self) -> &CStr {
+        // Safety: this is safe as initally len will be zero, resulting in an empty CStr;
+        // otherwise, the fields are initialized from init_args() from a pre-existing valid CStr.
+        unsafe { CStr::from_bytes_with_nul_unchecked(&self.buf[..self.len]) }
+    }
+}
+
+lazy_static! {
+    static ref ARGS: AtomicRefCell<StaticCString> = Default::default();
+}
```

Comment:
Would it work to just use plain [mut static](https://doc.rust-lang.org/reference/items/static-items.html#mutable-statics)?

```rust
static mut ARGS: [u8; 512];
```

Though that would require unsafe every time, so maybe not.

Reply:
I went back and forth between `lazy_static` and plain `static mut` several times, trying to reduce the amount of unsafe code as much as possible. With the `StaticCString` I think `lazy_static` made sense.

Now that I've switched over to `ArrayString`, I think the tables have turned: just plain `static mut` is good enough and more readable.